feat/added_ignore_dos_flag_to_mft_plugin

B0TAxy opened the pull request:
This change adds a new --ignore-dos flag to the MFT plugin. This allows users to filter out short DOS 8.3 filenames from the output, focusing only on long filenames. The update adds the argument to the CLI, updates the records method signature to handle the new boolean flag, and ensures the path resolution logic respects this setting when generating file paths.

B0TAxy committed:
Added flag

Schamper reviewed (changes requested):
Would you mind adding a small unit test? I believe we have a small MFT in the test data, I hope it has DOS names.

B0TAxy committed:
Merge branch 'fox-it:main' into feat/added_ignore_dos_flag_to_mft_plugin

B0TAxy committed:
Need to fix tests

B0TAxy committed:
Fixed tests need to add new ignore_dos tests

B0TAxy committed:
Added dos name test

B0TAxy committed:
Merge branch 'fox-it:main' into feat/added_ignore_dos_flag_to_mft_plugin

B0TAxy committed:
Merge branch 'fox-it:main' into feat/added_ignore_dos_flag_to_mft_plugin

B0TAxy commented:
> > > > It's fine to keep this open. When you have more information available you can open a issue or PR on dissect.ntfs.
> > > 
> > > I’ve actually tracked down the root cause—the bug was in how records were being handled, which was leading to both missed records and a lot of duplicates.
> > > 
> > > To fix this, I’ve refactored the entire plugin to use a generic function, eliminating the redundant logic we had before. I also took the opportunity to clean things up by removing the functions that were flagged for deletion in the previous version
> > 
> > Can you please detail what the bug was and what exactly you did to address it? Will the bug show up if I run the old code with the new test data?
> > 
> > Knowing that will making reviewing such large changes a lot easier.
> 
> Thank you for the follow-up. I understand that the scope of the changes is large, so here is a breakdown of the specific logic error and how the refactor addresses it.  
> The issue resided in the way ``iter_records`` handled record paths. While the ``ignore_dos`` flag was being passed to the top-level function (Example: records func), it was never actually utilized or passed down to the internal logic that iterated over the filename attributes.  
> The Result was code would process DOS filenames regardless of the flag, leading to the duplicates and missed records observed.  
> To answer your question: Yes, the bug will show up if you run the old code with the new test data. Since the previous implementation completely ignored the flag, it is unable to handle the new data requirements correctly.  
> To fix this: I updated the record iteration to properly respect the ``ignore_dos`` parameter, ensuring it filters filenames as intended.  
> Also, I noticed significant code duplication where multiple functions were performing nearly identical tasks. I consolidated these into a single generic function.  
> Lastly, I removed several legacy functions that were previously flagged for deletion, which streamlined the plugin but increased the number of lines changed in this PR.  
> This refactor not only fixes the immediate bug but also makes the plugin much easier to maintain by removing that redundant logic

I also wanted to clarify a specific side effect of the old logic: because ``iterate_records`` was being called for every path without the proper filtering, it would yield every single record regardless of context. This effectively meant the same records were being processed and yielded multiple times, causing the duplicated output.

B0TAxy committed:
Merge branch 'fox-it:main' into feat/added_ignore_dos_flag_to_mft_plugin

B0TAxy commented on dissect/target/plugins/filesystem/ntfs/mft.py:
```diff
@@ -491,6 +396,79 @@ def default_formatter(
+    return (
+        f"{md5}|{path}|{segment}|{file_mode}|{attr.owner_id}|{attr.security_id}|"
+        f"{filesize}|{attr.last_access_time}|{attr.last_modification_time}|"
+        f"{attr.last_change_time}|{attr.creation_time}"
```
Resolve + added test

B0TAxy committed:
CR fixes

B0TAxy commented on tests/plugins/filesystem/ntfs/test_mft.py:
```diff
@@ -165,8 +170,8 @@ def test_volume_identifier_none_guid() -> None:
-    outputs = [entry for entry in target_win_mft.mft.timeline() if re.match(regex_pattern, entry)]
+    """Test whether the MFT timeline functions as intended, same with the output."""
+    outputs = [entry for entry in target_win_mft.mft.timeline(ignore_dos=True) if re.match(regex_pattern, entry)]
 
```
I don't we need to test in every location if the dos flag works @Schamper what do you think?

B0TAxy commented:
Hi @Schamper, I’ve addressed the Copilot suggestions, except for a few cases where the automated logic didn't quite fit the context. Could you please take a look?

B0TAxy commented:
Hi @Schamper, just giving this a gentle bump to bring it back to your radar.

I know this is a pretty massive PR with a lot of moving parts and significant refactoring, so I completely understand it takes some time to digest. I believe all the previous discussion points have been addressed, and the tests are green.

Whenever someone has the cycles to tackle a larger review, I’d love to get your thoughts or see what else we need to do to get this across the finish line. Let me know if you want me to clarify or break down any specific refactored sections! Thanks!

B0TAxy committed:
Merge branch 'main' into feat/added_ignore_dos_flag_to_mft_plugin

B0TAxy committed:
Merge branch 'fox-it:main' into feat/added_ignore_dos_flag_to_mft_plugin
